zipline/utils/exchange_calendar.py

```diff
@@ -0,0 +1,386 @@
+#
+# Copyright 2016 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import (
+    ABCMeta,
+    abstractproperty,
+    abstractmethod,
+)
+from six import with_metaclass
+
+import pandas as pd
+from pandas import (
+    DataFrame,
+    date_range,
+    DateOffset,
+    DatetimeIndex,
+    Timedelta,
+)
+from pandas.tseries.offsets import CustomBusinessDay
+
+start_default = pd.Timestamp('1990-01-01', tz='UTC')
+end_base = pd.Timestamp('today', tz='UTC')
+# Give an aggressive buffer for logic that needs to use the next trading
+# day or minute.
+end_default = end_base + pd.Timedelta(days=365)
+
+
+def delta_from_time(t):
```

Comment:
I think Scotty had planned on using this on his branch but it's not necessary for ours: it can be deleted.